### PR TITLE
[Feature] Extend useEcosystem hook and update Layout and Ecosystem components

### PR DESCRIPTION
**Pull request by jnsdls**
## PR-Codex overview
This PR refactors ecosystem creation and management in the dashboard. 

### Detailed summary
- Updated ecosystem creation form UI
- Modified ecosystem status handling
- Improved ecosystem selection dropdown
- Removed unnecessary imports
- Enhanced error handling for ecosystem operations

> The following files were skipped due to too many changes: `apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[slug]/(active)/components/client/ecosystem-header.client.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jnsdls** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3604?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3604** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3604?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3603** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3603?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jnsdls and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by jnsdls** [comment]
this is blocked on ecosystem wallet API accepting `Authorization` header as auth (instead of only cookie)

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[slug]/(active)/hooks/use-ecosystem.ts [hunk 1]
```diff
@@ -8,15 +8,20 @@ export function useEcosystem({
...
+  initialData?: Ecosystem;
+}) {
   const { isLoggedIn } = useLoggedInUser();
 
   const ecosystemQuery = useQuery({
     queryKey: ["ecosystems", slug],
     queryFn: async () => {
       const res = await fetch(
         `${THIRDWEB_API_HOST}/v1/ecosystem-wallet/${slug}`,
-        { credentials: "include" },
```
Comment (joaquim-verges) [inline apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[slug]/(active)/hooks/use-ecosystem.ts:19]: No auth needed here?

Comment (joaquim-verges) [reply apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[slug]/(active)/hooks/use-ecosystem.ts:19]: I see a fetchEcosystem function below that hits the same endpoint, with the auth header. Prob want to reuse that here

**Review by gregfromstl — APPROVED** [review]

**Pull request merged**